cannot install dev-2024-06 (ubuntu)

**Describe the bug**

Using this plugin with mise and trying to upgrade from `dev-2024-05` to `dev-2024-06`. I get the following install error:

```sh
[DEBUG] $ ~/.local/share/mise/plugins/odin/bin/install
chmod: cannot access '/home/andrewchou/.local/share/mise/installs/odin/dev-2024-06/bin/odin': No such file or directory
asdf-odin: Expected /home/andrewchou/.local/share/mise/installs/odin/dev-2024-06/bin/odin to be executable.
asdf-odin: An error occurred while installing odin dev-2024-06.
[ERROR] ~/.local/share/mise/plugins/odin/bin/install failed
Error:
   0: ~/.local/share/mise/plugins/odin/bin/install exited with non-zero status: exit code 1
```

My best guess is that it seems like the zip directory structure is slightly different from the previous release.  The previous release has a `dist.zip` directory when unzipping, while this new one has `dist`.

![image](https://github.com/jtakakura/asdf-odin/assets/18542095/173827df-5bbf-4f01-89c4-fa98024ba825)

**Steps to reproduce**

I'm using mise, but probably reproducible with asdf:

```sh
mise install odin@dev-2024-06
```

**Expected behavior**

Installation should work without issue

--- a/.github/workflows/build.yml
+++ b/.github/workflows/build.yml
@@ -17,12 +17,6 @@ jobs:
   plugin_test_macos:
     runs-on: macos-latest
     steps:
-      - name: Download LLVM and setup PATH
-        run: |
-          brew install llvm@11
-          echo "/usr/local/opt/llvm@11/bin" >> $GITHUB_PATH
-          TMP_PATH=$(xcrun --show-sdk-path)/user/include
-          echo "CPATH=$TMP_PATH" >> $GITHUB_ENV
       - name: asdf_plugin_test
         uses: asdf-vm/actions/plugin-test@v1
         with:

--- a/lib/utils.bash
+++ b/lib/utils.bash
@@ -61,6 +61,8 @@ install_version() {
 
   if [ -d "$ASDF_DOWNLOAD_PATH/${platform}"_artifacts ]; then
     download_path="$ASDF_DOWNLOAD_PATH/${platform}_artifacts"
+  elif [ -d "$ASDF_DOWNLOAD_PATH/dist" ]; then
+    download_path="$ASDF_DOWNLOAD_PATH/dist"
   else
     download_path="$ASDF_DOWNLOAD_PATH"
   fi

